phing/tasks/tests.xml

```diff
@@ -83,21 +83,45 @@
   <target name="tests:phpunit" description="Executes PHPUnit tests against custom code.">
     <delete dir="${reports.localDir}/phpunit" quiet="true"/>
     <mkdir dir="${reports.localDir}/phpunit"/>
+    <if>
+      <and>
+        <isset property="phpunit.local" />
+        <not><isset property="env.TRAVIS" /></not>
+      </and>
+      <then>
+        <property name="phpunit.dirs" value="${phpunit.local}" />
+      </then>
+
+      <elseif>
+        <and>
+          <isset property="phpunit.deploy" />
+          <isset property="env.TRAVIS" />
+        </and>
+        <then>
+          <property name="phpunit.dirs" value="${phpunit.deploy}" />
+        </then>
+      </elseif>
+
+      <else>
+        <property name="phpunit.dirs" value="${repo.root}/tests/phpunit,${blt.root}/tests/phpunit/BltProject" />
+      </else>
+    </if>
+    <foreach list="${phpunit.dirs}" param="dir" target="tests:phpunit:dir"/>
+  </target>
 
+  <target name="tests:phpunit:dir" description="Run phpunit against given directories." hidden="true">
     <phpunit printsummary="true" failureproperty="phpunit.failures" errorproperty="phpunit.errors">
       <formatter type="plain" usefile="false"/>
       <formatter type="xml" todir="${reports.localDir}/phpunit"/>
       <batchtest>
-        <fileset dir="${repo.root}/tests/phpunit">
+        <fileset dir="${dir}">
```

Comment:
@grasmash I like this idea. I've updated the pull request so that it includes a few new targets to give us group support to phpunit.

Following your recommendation you can now define groups in the phpunit hash  in project.yml

```
phpunit:
  local:
     groups:
         - module
```

Unfortunately the PHPUnit phing task doesn't accept all configuration values defined in a phpunit.xml file, some information can be [found here](https://www.phing.info/trac/ticket/1233) and the [phpunit task docs](https://www.phing.info/docs/stable/hlhtml/index.html#PHPUnitTask) which suggest that only bootstrap, processIsolation, stopOnFailure, stopOnError, stopOnIncomplete and stopOnSkipped are supported.

So I couldn't use the phing task to execute the Drupal groups (this requires configuration from phpunit.xml.dist in `docroot/core`. To work around we can execute phpunit directly and pass in the required options.

So the approach is; I've defined 2 targets, `local:tests:phpunit` and `ci:tests:phpunit` which read values from *project.yml* (defined as above). They execute the `tests:phpunit:groups` target which is expecting the `phpunit.groups` and `phpunit.excludeGroups` properties. This allows us to execute group tests like:

```
blt local:tests:phpunit
blt ci:tests:phpunit
blt tests:phpunit:groups -Dphpunit.groups=module -Dphpunit.excludeGroups=module2
```